### DISCUSSION TITLE
Micromamba Hook not found

### DISCUSSION THREAD

**Discussion by Nationatori** [oobabooga/text-generation-webui#966] [Q&A]
![InkedScreenshot 2023-04-09 184613_LI](https://user-images.githubusercontent.com/102382763/230785948-f53fef39-f7d2-4f5c-8406-0f0a7de8b147.jpg)
hey guys,

i got an error and it says after the installation MicroMamba hook not found what should i do about that?

**Comment by handsomejack87** [comment]
maybe your path have non English characters?

**Reply by Nationatori** [reply]
Sorry for the late response,

I searched and ckecked it out if i can find a solution but no it told every time the same error and yes maybe it has to do with the non english characters but i don't know exactly what it causes.

**Comment by jllllll** [comment]
It could be either non English characters or having a space in the file path. The last time someone had the package cache error, I recommended that the use this alternative installer: https://github.com/jllllll/one-click-installers/tree/oobabooga-windows-miniconda

They said it worked for them.

**Reply by Nationatori** [reply]
Ok thanks i try that one

**Comment by tatelax** [comment]
Also having this. The alternative installer gives me a different error which says Miniconda not found.

**Comment by AlgorithmicOpposite** [comment]
I'm having the same error on my Windows 10.  "MicroMamba hook not found"  I was using Oobabooga with no problems until I tried to add a new model and edit some .bats lightly like instructed.  Is it possible to just uninstall anaconda and all the associated files?  I can't figure out how to get micromamba back on track!

**Reply by jllllll** [reply]
Use one of these:
https://github.com/jllllll/one-click-installers
https://github.com/xNul/one-click-installers/tree/universal

**Reply by AlgorithmicOpposite** [reply]
> Use one of these: https://github.com/jllllll/one-click-installers https://github.com/xNul/one-click-installers/tree/universal

I just tried both and get the same message, "Miniconda hook not found" and press any key to continue, that shuts down the window.
Thank you for trying!

**Reply by jllllll** [reply]
It is a completely new installer. You need to fully reinstall using one those. They are not compatible with the micromamba installer.

**Reply by AlgorithmicOpposite** [reply]
i think I did fully install with each one.  I'm not sure i know what you mean by fully reinstall?

**Reply by jllllll** [reply]
I mean delete text-generation-webui and installer_files. Another thing to keep in mind is path length. Unless you enable long path support in Windows, you need to install the webui close to the root of your drive. Something like `D:\textgen\`. Conda generates pretty long paths which Windows does not support by default for some inane reason. Also, I've heard that paths containing non-english characters can mess up Conda as well.

**Reply by AlgorithmicOpposite** [reply]
Ok thank you, I will work on that in the morning!  This problem is very aggravating lol.

**Reply by Nationatori** [reply]
I tried the miniconda version but i got the same problem like AlgorithmicOpposite and i also tried it with an short path like E:\textgen\ still got that error.

**Reply by AlgorithmicOpposite** [reply]
So, Nationatori, you haven't been able to fix it since you made this post?   jIIIII, you said the two installers you shared are not compatible with the micromamba installer.  Do you know why i would still be getting the same Miniconda error?

**Reply by jllllll** [reply]
@AlgorithmicOpposite It means that the script can't find the Miniconda files that it needs to activate the virtual environment. Typically, this is caused by the Miniconda installer failing to install. I honestly don't know why this would be happening and can only assume it is something particular with your system. Check the folders in `installer_files` for anything that is missing.

**Reply by Andie-Squirrel** [reply]
> @AlgorithmicOpposite It means that the script can't find the Miniconda files that it needs to activate the virtual environment. Typically, this is caused by the Miniconda installer failing to install. I honestly don't know why this would be happening and can only assume it is something particular with your system. Check the folders in `installer_files` for anything that is missing.

I tried fresh installing Miniconda and reinstalling oobabooga using the one-click-installer and official build and it still has issues with Miniconda.
![image](https://github.com/oobabooga/text-generation-webui/assets/119384539/49525f23-3ba7-4619-8945-eeb0a8a9c670)

I don't quite understand what it means by a path length that's "too long". Any help is greatly appreciated!

**Reply by Andie-Squirrel** [reply]
Oh! Did any of you have luck figuring out the problem?! I bet this must have been frustrating. Oddly enough, oobabooga was working for me yesterday and after uninstalling a couple Torch libraries and forgetting how to reinstall all of them, Miniconda started flipping out haha.

**Reply by jllllll** [reply]
> > @AlgorithmicOpposite It means that the script can't find the Miniconda files that it needs to activate the virtual environment. Typically, this is caused by the Miniconda installer failing to install. I honestly don't know why this would be happening and can only assume it is something particular with your system. Check the folders in `installer_files` for anything that is missing.
> 
> I tried fresh installing Miniconda and reinstalling oobabooga using the one-click-installer and official build and it still has issues with Miniconda. ![image](https://user-images.githubusercontent.com/119384539/238087680-49525f23-3ba7-4619-8945-eeb0a8a9c670.png)
> 
> I don't quite understand what it means by a path length that's "too long". Any help is greatly appreciated!

Windows has a limit to the number of characters that can be in a file path (256 characters). There is a registry setting that can be changed to greatly increase this limit. The installer installs a copy of Miniconda for it to use. Their installer currently does not allow you to install Miniconda under a long path in silent mode, which is what we use. I included the warning because the Miniconda installer will just fail and not tell you why, which can be pretty confusing.

Your comment about reinstalling Miniconda leads me to believe that you are trying to run the one-click-installer under an active Conda environment. This works fine for some people, but can lead to any number of issues with the webui as you will end up nesting multiple Conda environments in an unconventional way.

**Reply by Andie-Squirrel** [reply]
> > > @AlgorithmicOpposite It means that the script can't find the Miniconda files that it needs to activate the virtual environment. Typically, this is caused by the Miniconda installer failing to install. I honestly don't know why this would be happening and can only assume it is something particular with your system. Check the folders in `installer_files` for anything that is missing.
> > 
> > 
> > I tried fresh installing Miniconda and reinstalling oobabooga using the one-click-installer and official build and it still has issues with Miniconda. ![image](https://user-images.githubusercontent.com/119384539/238087680-49525f23-3ba7-4619-8945-eeb0a8a9c670.png)
> > I don't quite understand what it means by a path length that's "too long". Any help is greatly appreciated!
> 
> Windows has a limit to the number of characters that can be in a file path (256 characters). There is a registry setting that can be changed to greatly increase this limit. The installer installs a copy of Miniconda for it to use. Their installer currently does not allow you to install Miniconda under a long path in silent mode, which is what we use. I included the warning because the Miniconda installer will just fail and not tell you why, which can be pretty confusing.
> 
> Your comment about reinstalling Miniconda leads me to believe that you are trying to run the one-click-installer under an active Conda environment. This works fine for some people, but can lead to any number of issues with the webui as you will end up nesting multiple Conda environments in an unconventional way.

Thank you very much for the timely response and detailed explanation!! I'll look into that, I wasn't aware of such a registry setting.
Sorry if this sounds very dumb, but I installed it through the included batch script. I haven't run it using Anaconda Prompt so I assume it isn't activated.

I was concerned about the nested environments. If it isn't too much of a bother, could you recommend if I should uninstall a Conda installation if there's any sort of conflict with the packages or whatnot? Again, if it isn't too much of an issue, could you provide me a link to the registry setting? I am very unfamiliar with registry hacks and I don't want to do any further damage to my system.

Thanks again!

**Reply by jllllll** [reply]
@Andie-Squirrel If you aren't running Anaconda Prompt then you should be fine in that regard. The easiest way I know of to enable long paths is to open CMD as admin and enter this command:
```Batch
Reg add "HKLM\SYSTEM\CurrentControlSet\Control\FileSystem" /v "LongPathsEnabled" /t REG_DWORD /d "1" /f
```
I'm unsure if you need to restart your system afterwards for it to take affect.

I've seen people report problems with the installer when having Miniconda already installed on their system, but I'm unsure if this is the cause of your issue. I have Miniconda installed on my system as well, but don't have any issues with the installer.

**Reply by Andie-Squirrel** [reply]
Wow, thank you so much.
I ran the instruction and thus far I'm still getting the same results. I'm downloading some models so I can't restart at the moment, but I'll come back later to report if it worked or not.

Again thank you so much. It could be an installer issue on my part, haven't the faintest clue why oobabooga stopped worked between yesterday and today except that I uninstalled and reinstalled torch.
Glad to hear your installer works correctly. Likely this is just a path issue to access Miniconda.

**Reply by Andie-Squirrel** [reply]
> @Andie-Squirrel If you aren't running Anaconda Prompt then you should be fine in that regard. The easiest way I know of to enable long paths is to open CMD as admin and enter this command:
> 
> ```batchfile
> Reg add "HKLM\SYSTEM\CurrentControlSet\Control\FileSystem" /v "LongPathsEnabled" /t REG_DWORD /d "1" /f
> ```
> 
> I'm unsure if you need to restart your system afterwards for it to take affect.
> 
> I've seen people report problems with the installer when having Miniconda already installed on their system, but I'm unsure if this is the cause of your issue. I have Miniconda installed on my system as well, but don't have any issues with the installer.

Hi again. I tried the script and it still seems to take issue with the path size. I'm not quite sure what to do at this point. The line of code you provided appeared to work correctly but not allow larger path sizes.

![GIF 5-13-2023 2-44-19 PM](https://github.com/oobabooga/text-generation-webui/assets/119384539/8e670b24-c529-4d07-bcc1-84eaeb20d762)


Or in CMD:
![DebugOobabooga](https://github.com/oobabooga/text-generation-webui/assets/119384539/76d893f9-c8b7-4830-b388-d91b74af406c)

Maybe the issue lies with the fact I have Oobabooga installed on a different drive than Miniconda?
I'll try using symbolic links, maybe that oughta fix the problem.

**Comment by Nationatori** [comment]
Hello guys, i checked it out and it took me some time that i found out there is a folder missing, named text-generation-webui and it won't install it, and it also makes a folder of my *Pc account name* root folder and still didn't figure it out how to fix that, but it does install the installer_files.

**Comment by Nationatori** [comment]
New update fixed it
